### ISSUE TITLE
enhancment Allow for modules to provide there own c files

### ISSUE THREAD

**Issue by Nosh-Ware** [Nuitka/Nuitka#1593]
I'm not sure how viable this would be, but one big issue for me is seeing how long a recursive build of a script that uses alot of libraries, most being unities that could be simplified if a hand written c file was substituted in for them. Would it be possible to support come, maybe comment markup in a python module for accessing a .c file that can be substituted in for nukita trying to compile that library?

**Comment by kayhayen** [comment]
If what you want is caching, well that is implemented separately.

**Comment by Nosh-Ware** [comment]
Ah, seems like I'm going to have to do my homework then. Currently Nukita is just a magic box.

**Comment by kayhayen** [comment]
Yes, caching. Replacing a module with C code of your own will be practically be impossible, so what I am assuming you want to do, is caching. I can need help there, but there is issues with the XML persistent of variable scopes that I would have to resolve first.

**Comment by Nosh-Ware** [comment]
I have no Idea what scopes are, but my idea was to superseded any kind of automated parsing of a module and provide some type of comment markup directly linking Py functions to C functions and Py variables to C variables, perhaps with (again, not sure how to do this but I have heard its possible) having static types enforced on the python side? I understand that this eliminates a ton of functionality in python but in my case all I want to add is a few functions. I can take them out of the class, add some comments to direct nukita what file tor reference and manually link the functions and variables, and call it done.

**Comment by kayhayen** [comment]
That's a great idea, looking forward to seeing an implementation of it.

**Comment by Nosh-Ware** [comment]
Well, I'll try my best then.
